python/whylogs/experimental/extras/confighub.py

```diff
@@ -0,0 +1,147 @@
+import logging
+import os
+from abc import ABC, abstractmethod
+from typing import List, Optional
+
+from semantic_version import Version
+
+logger = logging.getLogger(__name__)
+
+
+class ConfigStore(ABC):
+    def __init__(self, org_id: str, dataset_id: str, what: str, create: bool = False):
+        """
+        what in {"constraints", "validators", ... }
+        """
+        self._org_id = org_id
+        self._dataset_id = dataset_id
+        self._what = what
+        if create and not self.exists():
+            self.create()
+
+    @abstractmethod
+    def exists(self) -> bool:
+        """Returns True iff the store exists"""
+        return False
+
+    @abstractmethod
+    def create(self) -> None:
+        """Create the store if it doesn't already exist"""
+        pass
+
+    @abstractmethod
+    def get_latest(self) -> str:
+        """Returns latest version of the config"""
+        pass
+
+    def get_version_of_latest(self) -> Version:
+        """
+        Returns the Version of the current config
+        """
+        return self.get_available_versions()[-1]
+
+    @abstractmethod
+    def get_version(self, ver: Version) -> str:
+        """
+        Returns the requested version of the config. ValueError on invalid/unavailable.
+        """
+        pass
+
+    @abstractmethod
+    def get_available_versions(self) -> List[Version]:
+        """
+        Returns available versions in ascending order
+        """
+        pass
+
+    @abstractmethod
+    def propose_version(self, config: str, new_version: Version, note: Optional[str] = None) -> None:
+        """
+        Submit config to review process with proposed new verison.
+        ValueError if new_version <= latest
+        """
+        pass
+
+    @abstractmethod
+    def commit_version(self, new_version: Version) -> None:
+        """
+        Make the proposed config the latest version. ValueError if proposed new_version doesn't exist
+        """
+        pass
+
+
+class LocalGitConfigStore(ConfigStore):
+    def _run_commands(self, commands: List[str]) -> None:
+        for command in commands:
+            output = os.popen(f"cd {self._path}; {command}")
```

Comment:
I should use the fancier `popen()` so I can capture (and ignore) the `stderr` output.

Reply:
I'm not sure how well this implementation will work on non-POSIX environments. For hackathon I don't care :)